src/components/Popup/OfficePopup.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+    <div
+            ref="self"
+            :class="classNames.root"
+            @focus="focused = true"
+            @blur="onSelfBlur"
+    >
+        <slot/>
+    </div>
+</template>
+
+<script lang="ts">
+    import {Component, Vue, Prop} from "vue-property-decorator";
+    import {mergeStyleSets} from "@uifabric/merge-styles";
+    import {getDocument} from "@/utility/dom";
+
+    @Component
+    export default class OfficePopup extends Vue {
+        private originalFocusedElement?: HTMLElement;
+        private focused: boolean = false;
+
+        get classNames() {
+            return mergeStyleSets({
+                root: [
+                    this.needsVerticalScrollBar && {
+                        overflowY: "scroll"
+                    },
+                    !this.needsVerticalScrollBar && {
+                        overflowY: undefined
+                    }]
+            });
+        }
+
+        @Prop({type: Boolean, default: true}) private needsVerticalScrollBar!: boolean;
```

Comment:
Move the Properties at the top of the class. Otherwise this looks ready to merge!